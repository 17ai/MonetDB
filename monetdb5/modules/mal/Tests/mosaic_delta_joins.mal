# Delta compression
b:= bat.new(:oid,:int);
bat.append(b,0);
bat.append(b,1);
bat.append(b,4);
bat.append(b,-1);
bat.append(b,-3);
bat.append(b,1000);
bat.append(b,1001);
bat.append(b,1004);
bat.append(b,999);
bat.append(b,977);
base:=algebra.copy(b);

io.print(base);
x:= mosaic.compress(b,"delta");

c:= bat.new(:oid,:oid);
bat.append(c,0@0);
bat.append(c,2@0);
bat.append(c,5@0);
io.print(c);

#mosaic.dump(x);
s:= algebra.leftfetchjoin(c,base);
io.print(s);
xs:= mosaic.leftfetchjoin(c,x);
io.print(xs);

d:= bat.new(:oid,:int);
bat.append(d,4);
bat.append(d,-1);
bat.append(d,nil:int);

(jl,jr):= algebra.join(base,d);
io.print(jl,jr);
(ml,mr):= mosaic.join(x,d);
io.print(ml,mr);

(kl,kr):= algebra.join(d,base);
io.print(kl);
io.print(kr);
io.print(kl,kr);
(nl,nr):= mosaic.join(d,x);
io.print(nl);
io.print(nr);
io.print(nl,nr);
