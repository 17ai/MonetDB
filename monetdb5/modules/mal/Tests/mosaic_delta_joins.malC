# Delta compression
b:= bat.new(:oid,:int);
bat.append(b,0);
bat.append(b,1);
bat.append(b,4);
bat.append(b,-1);
bat.append(b,-3);
bat.append(b,1000);
bat.append(b,1001);
bat.append(b,1004);
bat.append(b,999);
bat.append(b,977);
base:=algebra.copy(b);

io.print(base);
x:= mosaic.compress(b,"delta");

c:= bat.new(:oid,:oid);
bat.append(c,0@0);
bat.append(c,2@0);
bat.append(c,5@0);
io.print(c);

#mosaic.dump(x);
s:= algebra.projection(c,base);
io.print(s);
xs:= mosaic.projection(c,x);
io.print(xs);

d:= bat.new(:oid,:int);
bat.append(d,4);
bat.append(d,-1);
bat.append(d,nil:int);

(jl,jr):= algebra.subjoin(base,d, nil:bat, nil:bat,false,0:lng);
io.print(jl,jr);
(ml,mr):= mosaic.subjoin(x,d, nil:bat, nil:bat,false,0:lng);
io.print(ml,mr);

(kl,kr):= algebra.subjoin(d,base, nil:bat, nil:bat,false,0:lng);
io.print(kl);
io.print(kr);
io.print(kl,kr);
(nl,nr):= mosaic.subjoin(d,x, nil:bat, nil:bat,false,0:lng);
io.print(nl);
io.print(nr);
io.print(nl,nr);
