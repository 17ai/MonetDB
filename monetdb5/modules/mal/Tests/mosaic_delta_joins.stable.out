stdout of test 'mosaic_delta_joins` in directory 'monetdb5/modules/mal` itself:


# 19:38:40 >  
# 19:38:40 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=30180" "--set" "mapi_usock=/var/tmp/mtest-26536/.s.monetdb.30180" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/export/scratch1/mk/mosaic//Linux/var/MonetDB/mTests_monetdb5_modules_mal" "mosaic_delta_joins.mal"
# 19:38:40 >  

# MonetDB 5 server v11.20.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_modules_mal', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 15.591 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2014 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.ins.cwi.nl:30180/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-26536/.s.monetdb.30180
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
function user.main():void;
# Delta compression 
    b := bat.new(:oid,:int);
    bat.append(b,0);
    bat.append(b,1);
    bat.append(b,4);
    bat.append(b,-1);
    bat.append(b,-3);
    bat.append(b,1000);
    bat.append(b,1001);
    bat.append(b,1004);
    bat.append(b,999);
    bat.append(b,977);
    base := algebra.copy(b);
    io.print(base);
    x := mosaic.compress(b,"delta");
    c := bat.new(:oid,:oid);
    bat.append(c,0@0);
    bat.append(c,2@0);
    bat.append(c,5@0);
    io.print(c);
#mosaic.dump(x); 
    s := algebra.leftfetchjoin(c,base);
    io.print(s);
    xs := mosaic.leftfetchjoin(c,x);
    io.print(xs);
    d := bat.new(:oid,:int);
    bat.append(d,4);
    bat.append(d,-1);
    bat.append(d,nil:int);
    (jl,jr) := algebra.join(base,d);
    io.print(jl,jr);
    (ml,mr) := mosaic.join(x,d);
    io.print(ml,mr);
    (kl,kr) := algebra.join(d,base);
    io.print(kl);
    io.print(kr);
    io.print(kl,kr);
    (nl,nr) := mosaic.join(d,x);
    io.print(nl);
    io.print(nr);
    io.print(nl,nr);
end user.main;
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	0  ]
[ 1@0,	1  ]
[ 2@0,	4  ]
[ 3@0,	-1  ]
[ 4@0,	-3  ]
[ 5@0,	1000  ]
[ 6@0,	1001  ]
[ 7@0,	1004  ]
[ 8@0,	999  ]
[ 9@0,	977  ]
#header block 276e0e0 version 20140808
#index top 1
#[0] 5 96
#header block 276e0e0 version 20140808
#index top 2
#[0] 5 96
#[1] 10 120
#blk  type 5 todo 0
#wins 0 0 0 2 0 0 
#elms 0 0 0 10 0 0 
#time 0 0 0 0 0 0 
#perc 0 0 0 100 0 0 
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	2@0  ]
[ 2@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	0  ]
[ 1@0,	4  ]
[ 2@0,	1000  ]
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	0  ]
[ 1@0,	4  ]
[ 2@0,	1000  ]
#--------------------------#
# h	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	2@0,	0@0  ]
[ 1@0,	3@0,	1@0  ]
#--------------------------#
# h	t	t  # name
# void	void	void  # type
#--------------------------#
[ 0@0,	0@0,	7@0  ]
[ 1@0,	2@0,	0@0  ]
[ 2@0,	3@0,	1@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	1@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	3@0  ]
#--------------------------#
# h	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	0@0,	2@0  ]
[ 1@0,	1@0,	3@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	7@0  ]
[ 1@0,	0@0  ]
[ 2@0,	1@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	2@0  ]
[ 2@0,	3@0  ]
#--------------------------#
# h	t	t  # name
# void	void	void  # type
#--------------------------#
[ 0@0,	7@0,	0@0  ]
[ 1@0,	0@0,	2@0  ]
[ 2@0,	1@0,	3@0  ]

# 19:38:40 >  
# 19:38:40 >  "Done."
# 19:38:40 >  

